pontoon/actionlog/migrations/0004_auto_20240816_1045.py

```diff
@@ -0,0 +1,48 @@
+# Generated by Django 4.2.11 on 2024-08-16 10:45
+
+import datetime
+
+from django.db import migrations
+
+
+def migrate_translation_to_actionlog(apps, schema_editor):
+    Translation = apps.get_model("base", "Translation")
+    ActionLog = apps.get_model("actionlog", "ActionLog")
+
+    traslation_info = {
+        "date": ActionLog.ActionType.TRANSLATION_CREATED,
+        "approved_date": ActionLog.ActionType.TRANSLATION_APPROVED,
+        "unapproved_date": ActionLog.ActionType.TRANSLATION_UNAPPROVED,
+        "rejected_date": ActionLog.ActionType.TRANSLATION_REJECTED,
+        "unrejected_date": ActionLog.ActionType.TRANSLATION_UNREJECTED,
+    }
+
+    # date to end migration
+    end_date = datetime.datetime(
+        2020, 1, 7, 9, 25, 11, 829125, tzinfo=datetime.timezone.utc
+    )
+
+    actions_to_log = []
+
+    for translation in Translation.objects.filter(date__lt=end_date):
+        for attr, action_type in traslation_info.items():
+            if getattr(translation, attr) is not None:
+                actions_to_log.append(
+                    ActionLog(
+                        action_type=action_type,
+                        created_at=translation.date,
+                        entity_id=translation.entity_id,
+                        locale_id=translation.locale_id,
+                        performed_by_id=translation.user_id,
+                        translation_id=translation.id,
+                    )
+                )
+    ActionLog.objects.bulk_create(actions_to_log)
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("actionlog", "0003_existing_pretranslation_action"),
+    ]
+
+    operations = []
```

Comment:
We should call the function from here, as well as the reverse migration ([RunPython.noop](https://docs.djangoproject.com/en/5.1/ref/migration-operations/#django.db.migrations.operations.RunPython.noop) should be fine).